### PR TITLE
O auth revoke

### PR DESCRIPTION
**Pull request by haaakon**
Adds handling of http unauthorized error when doing requests. Will revoke current access token and try again.

### CODE REVIEW

File: Digipost/APIClient+URLSessionTask.swift [hunk 1]
```diff
@@ -109,17 +114,24 @@ extension APIClient {
...
                 }
                 // we get here if the request was canceled, should do nothing.
         })
         return task
     }
 
-    func isUnauthorized(urlResponse: NSHTTPURLResponse?) -> Bool {
+    private func isUnauthorized(urlResponse: NSHTTPURLResponse?) -> Bool {
         if let actualResponse = urlResponse as NSHTTPURLResponse! {
             if (actualResponse.statusCode == 403 || actualResponse.statusCode == 401) {
```
Comment (lillesand) [inline Digipost/APIClient+URLSessionTask.swift:137]: I think we should separate between 403 (Forbidden) and 401 (Unauthorized). 401 states that the user is not authorized (or rather: authenticated), 403 status that the user is authenticated, but not allowed to access the given resource. In this pull requests we want to be looking for 401.

File: Digipost/APIClient+URLSessionTask.swift [hunk 2]
```diff
@@ -137,32 +157,42 @@ extension APIClient {
             urlRequest.setValue(value, forHTTPHeaderField: key)
         }
 
-        let task = jsonDataTask(urlRequest, success: success, failure: failure)
+        let task = jsonDataTask(urlRequest, success: success) { (error) -> () in
+            if error.code == Constants.Error.Code.oAuthUnathorized.rawValue {
+                OAuthToken.removeAcessTokenForOAuthTokenWithScope(kOauth2ScopeFull)
```
Comment (lillesand) [inline Digipost/APIClient+URLSessionTask.swift]: Small typo: removeA`c`essToken

File: Digipost/APIClient+URLSessionTask.swift (hunk 1 above)
Comment (haaakon) [inline Digipost/APIClient+URLSessionTask.swift:137]: commit: b2f408142038f69b50b4e59823d63548b72f840d

File: Digipost/APIClient+URLSessionTask.swift (hunk 2 above)
Comment (haaakon) [inline Digipost/APIClient+URLSessionTask.swift]: commit: 869a23700d9591bb01b4be07907150306ee58a1c

**Comment by lillesand** [comment]
Looks good, merge when ready!

**Pull request merged**